tests/cypress/integration/IKUpload.cy.js

```diff
@@ -0,0 +1,39 @@
+describe('IKUpload Element', () => {
+    const APP_HOST = Cypress.env().APP_HOST;
+
+    describe('Image Upload', () => {
+        it('should upload image and render same image', () => { 
+            //static file
+            const p = 'sample.jpeg'
+
+            // launch URL
+            cy.visit(APP_HOST);
+
+            //upload file with attachFile
+            cy.get('.file-upload-ik').attachFile(p)
+
+            // wait for 2 secs
+            cy.wait(2000);
+
+            //Verify uploaded file
+            cy.get('.uploaded-img-ik').should('have.attr', 'src')
```

Comment:
The aim of this test suite is assert that `<IKUpload>` props work as we have mentioned in the documentation. Among those props are:
* `inputRef`
* `onUploadStart`
* `onUploadProgress`
* `validateFile`
* `onSuccess`
* `onError`

Anyways `src` attribute will always be there. Asserting on value is not sufficient. What we should do is set some global variable in `onSuccess` function and other prop methods e.g. `windows.onSuccessCalled = true`, and then in test cases we should assert on its value. Also right before calling `cy.get('.file-upload-ik').attachFile(p)` we should assert the negative i.e. `windows.onSuccessCalled` is indeed undefined.

Reply:
As discussed, in cypress, we're not covering corner cases.